Parse sub-groups correctly

robert-annett opened the pull request:
I have extracted the creation of the log group into its own method so it
can be called iteratively. The sub-data dictionary is also passed
through so that the groups can be found within it.

robert-annett committed:
Parse sub-groups correctly

I have extracted the creation of the log group into its own method so it
can be called iteratively. The sub-data dictionary is also passed
through so that the groups can be found within it.

briancoyner commented:
@robert-annett can you add a sample log file containing sub-groups. It appears that I was collecting various log files under `src/test/resources/logs`, so feel free to put the sample log file there.

robert-annett committed:
Unit test with nested groups

This commit contains a FIX message that has nested groups based on
side->parties->subparties. This is within an FXSpot execution report.

There has also been a minor change to the test class so that nested
fields are counted and checked for.

robert-annett commented:
@briancoyner Sample log file is now included and unit test modified slightly so it counts fields in sub-groups for any tests. Note that this is in a second commit in the same PR.

briancoyner commented:
@robert-annett I'll get it reviewed/ merged soon (next few days or so). Thanks.
